store.go

```diff
@@ -118,3 +207,168 @@ func (b *BucketReaderAt) ReadAt(p []byte, off int64) (n int, err error) {
+			switch e := expr.(type) {
+			case *logicalplan.Column:
+				leftColumnRef = &ColumnRef{
+					ColumnName: e.ColumnName,
+				}
+				return false
+			}
+			return true
+		}))
+		if leftColumnRef == nil {
+			return nil, nil, nil, errors.New("left side of binary expression must be a column")
+		}
+
+		min, minok := mins[leftColumnRef.ColumnName]
+		max, maxok := maxes[leftColumnRef.ColumnName]
+		if !minok && !maxok { // column doesn't exist
+			return nil, nil, nil, ErrColNotExist
+		}
+
+		var rightValue parquet.Value
+		e.Right.Accept(PreExprVisitorFunc(func(expr logicalplan.Expr) bool {
+			switch e := expr.(type) {
+			case *logicalplan.LiteralExpr:
+				rightValue, err = pqarrow.ArrowScalarToParquetValue(e.Value)
+				return false
+			}
+			return true
+		}))
+		if err != nil {
+			return nil, nil, nil, err
+		}
+
+		// Create parquet.Value of our min and max
+		return min, max, &rightValue, nil
+	}
+
+	switch e := expr.(type) {
+	case *logicalplan.BinaryExpr:
+		switch e.Op {
+		case logicalplan.OpNotEq:
+			min, max, v, err := parseExpr(e)
+			if err != nil {
+				return false, err
+			}
+			return compare(v, max) == 1 || compare(v, min) == -1, nil
+		case logicalplan.OpLtEq:
+			min, _, v, err := parseExpr(e)
+			if err != nil {
+				return false, err
+			}
+			return compare(min, v) <= 0, nil
+		case logicalplan.OpLt:
+			min, _, v, err := parseExpr(e)
+			if err != nil {
+				return false, err
+			}
+			return compare(min, v) < 0, nil
+		case logicalplan.OpGt:
+			_, max, v, err := parseExpr(e)
+			if err != nil {
+				return false, err
+			}
+			return compare(max, v) > 0, nil
+		case logicalplan.OpGtEq:
+			_, max, v, err := parseExpr(e)
+			if err != nil {
+				return false, err
+			}
+			return compare(max, v) >= 0, nil
+		case logicalplan.OpEq:
+			min, max, v, err := parseExpr(e)
+			if err != nil {
+				return false, err
+			}
+			return compare(v, max) <= 0 || compare(v, min) >= -1, nil
+		case logicalplan.OpAnd:
+			left, lefterr := compareMinsMaxes(mins, maxes, e.Left)
+			if lefterr != nil && lefterr != ErrColNotExist {
+				return false, lefterr
+			}
+
+			right, righterr := compareMinsMaxes(mins, maxes, e.Right)
+			if righterr != nil && righterr != ErrColNotExist {
+				return false, righterr
+			}
+
+			return (left || lefterr != nil) && (right || righterr != nil), nil
+		case logicalplan.OpOr:
+			left, lefterr := compareMinsMaxes(mins, maxes, e.Left)
+			if lefterr != nil && lefterr != ErrColNotExist {
+				return false, lefterr
+			}
+
+			right, righterr := compareMinsMaxes(mins, maxes, e.Right)
+			if righterr != nil && righterr != ErrColNotExist {
+				return false, righterr
+			}
+
+			return left || right, nil
+		default:
+			// unsupported op; return true. Better to have false positives
+			return true, nil
+		}
+	default:
+		return false, fmt.Errorf("unsupported expr %T", e)
+	}
+}
+
+// compares two parquet values. 0 if they are equal, -1 if v1 < v2, 1 if v1 > v2.
+func compare(v1, v2 *parquet.Value) int {
```

Comment:
`parquet.Type` has a `Compare` method, I think you should be able to use that instead of this function

Reply:
ah ha! I knew there was something like that, thanks for pointing me to it!